tensorrt_llm/_torch/auto_deploy/shim/ad_executor.py

```diff
@@ -163,29 +166,47 @@ def _prepare_inputs(
         input_pos: List[int] = []
         last_logit_only: List[bool] = []
         page_assignments: List[List[int]] = []
+        previous_batch_indices: List[int] = []
+        # We keep track of the batch index for the requests.
+        # This is needed by the overlap scheduler.
+        batch_idx = 0
 
         # look at context requests first
         for request in context_requests:
             # store input ids and pos of first token in sequence
             input_ids.append(request.get_tokens(0))
             input_pos.append(request.context_current_position)
 
-            # only return last logit
+            request.py_batch_idx = batch_idx
+            batch_idx += 1
             last_logit_only.append(True)
 
         # look at extend+generate requests next
         for request in chain(extend_requests, gen_requests):
-            # store input ids and pos of first token in sequence
-            input_ids.append([request.get_token(0, request.get_num_tokens(0) - 1)])
-            input_pos.append(request.max_beam_num_tokens - 1)
+            # new_tokens are provided when the overlap scheduler is enabled.
+            if new_tokens is None or request.is_dummy or request.py_batch_idx is None:
+                input_ids.append([request.get_token(0, request.get_num_tokens(0) - 1)])
+                input_pos.append(request.max_beam_num_tokens - 1)
+            else:
+                previous_batch_indices.append(request.py_batch_idx)
```

Comment:
So this `request` object has a `py_batch_idx` and we store it here? But then below we also update it? So we somehow use the historical information here